src/index.tsx

```diff
@@ -23,6 +23,7 @@ WebFont.load({
 });
 
 const rootElement = document.getElementById("root");
+const queryClient = new QueryClient();
```

Comment:
Here is the main fix. The query client was being reinitialized when the `Root` component re-rendered. This would lead to multiple caches and weird bugs. Moving it to the module scope ensures only one instance exists.

Reply:
Fantastic!